### ISSUE TITLE
Feature request: Add eip-4337 Account Abstraction implementation

### ISSUE THREAD

**Issue by iam-dev** [solidstate-network/solidstate-solidity#202]
ERC-4337: Account Abstraction Using Alt Mempool  
An account abstraction proposal which completely avoids consensus-layer protocol changes, instead relying on higher-layer infrastructure.

Contracts:

- BaseAccount
- BasePaymaster
- EntryPoint
- SenderCreator
- StakeManager
- WalletBase
- WalletFactory
- UserOperations

**Comment by iam-dev** [comment]
I might do temporary pull request regularly just to have a discussions about the best way to implement this ...

**Comment by ItsNickBarry** [comment]
Yes, this is something I will look into soon.  Feel free to open a draft PR - might be helpful.

**Comment by tjvsx** [comment]
@iam-dev @ItsNickBarry there was an ETHDenver submission that combined eip4337 and eip2535 -- https://github.com/kopy-kat/ethdenver-aa/tree/main/rhinestone-contracts -- may be helpful

**Comment by tjvsx** [comment]
The whole account abstraction world seems to evolve more each day, so I think it may be too early to identify standard implementations. But I thought it was worth sharing [this brief attempt at EIP-6900](https://github.com/tjvsx/blind-eip-6900), which extends ERC4337. Ran into some issues that I don't have time to tackle right now, but thought it could be worth checking out for the next deep diver

**Comment by ItsNickBarry** [comment]
I think account abstraction will be my next priority.  Thanks @tjvsx, I'll look at your link when I start working on this.
